service/routers/decks.js

```diff
@@ -20,78 +21,71 @@ const getDecks = async (req, res) => {
-        return res.status(400).json({
-            error: errors.array()[0].msg,
-            param: errors.array()[0].param
-        })
-    }
-
-  if(req.user.role == "Admin" || req.user.role == "SuperUser" || req.user.userId == req.params.userId) {
-    const { userId, role } = req.user
-    const newDeck = req.body
+  const userId = req.user.userId
+  const newDeck = req.body
+  const requestor = await User.findById(userId)
+  if (
+    requestor.role === 'admin' ||
+    requestor.role === 'superuser' ||
+    requestor.role === 'user'
+  ) {
     try {
-      const user = await User.findById(req.params.userId)
-      user.decks.push({
+      requestor.decks.push({
         name: newDeck.name,
-        cards: []
+        cards: [],
       })
-      await user.save()
+      await requestor.save()
       res.sendStatus(204)
     } catch (err) {
       console.log(`${createDeck.name}: ${err}`)
       res.sendStatus(500)
     }
   } else {
-    res.status(404).send('you are not authorized to access this route')
+    res.sendStatus(403).send('forbidden')
   }
 }
 
 const createCard = async (req, res) => {
-  const errors = validationResult(req)
-
-    if(!errors.isEmpty()){
-        return res.status(400).json({
-            error: errors.array()[0].msg,
-            param: errors.array()[0].param
-        })
-    }
-
-  if(req.user.role == "Admin" || req.user.role == "SuperUser" || req.user.userId == req.params.userId) {
-    const { userId, role } = req.user
-    const deckId = req.params.deckId
-    const newCard = req.body
+  const userId = req.user.userId
+  const deckId = req.params.id
+  const newCard = req.body
+  const requestor = await User.findById(userId)
+  if (
+    requestor.role === 'admin' ||
+    requestor.role === 'superuser' ||
+    requestor.role === 'user'
+  ) {
     try {
-      const user = await User.findById(req.params.userId)
-      const deck = user.decks.id(deckId)
+      const deck = requestor.decks.id(deckId)
       deck.cards.push(newCard)
-      await user.save()
+      await requestor.save()
       const newId = deck.cards[deck.cards.length - 1]
       res.status(200).send(newId._id)
     } catch (err) {
       console.log(`${createCard.name}: ${err}`)
       res.sendStatus(500)
     }
   } else {
-    res.status(404).send('you are not authorized to access this route')
+    res.sendStatus(403).send('forbidden')
   }
-  
 }
 
 const deleteDeck = async (req, res) => {
-  if(req.user.role == "Admin" || req.user.role == "SuperUser" || req.user.userId == req.params.userId) {
-    const { userId, role } = req.user
-    const deckId = req.params.deckId
+  const userId = req.user.userId
+  const deckId = req.params.id
+  const requestor = await User.findById(userId)
+  if (
+    requestor.role === 'admin' ||
+    requestor.role === 'superuser' ||
+    requestor.role === 'user'
+  ) {
     try {
-      const user = await User.findById(req.params.userId)
+      const user = await User.findById(userId)
```

Comment:
Same comments as above: we need to guard against a requestor with a user role deleting other user's decks